kiali-server/templates/service.yaml

```diff
@@ -31,11 +31,21 @@ spec:
   - name: http
   {{- end }}
     protocol: TCP
+    {{- if .Values.deployment.service_port }}
+    targetPort: {{ .Values.server.port }}
+    port: {{ .Values.deployment.service_port }}
+    {{- else }}
     port: {{ .Values.server.port }}
+    {{- end }}
   {{- if .Values.server.metrics_enabled }}
   - name: http-metrics
     protocol: TCP
+    {{- if .Values.deployment.service_metrics_port }}
+    targetPort: {{ .Values.server.metrics_port }}
+    port: {{ .Values.deployment.service_metrics_port }}
+    {{- else }}
     port: {{ .Values.server.metrics_port }}
+    {{- end }}
```

Comment:
changes to this service.yaml template does not appear to be related to the issue of certificate creation. If so, this should be in a separate PR under a separate enhancement github issue.